### PR TITLE
adds cookie banner and settings functionality

### PR DESCRIPTION
**Pull request by desderry**

### CODE REVIEW

File: src/SFA.DAS.Campaign.Web/Views/Shared/_Layout-campaign.cshtml [hunk 1]
```diff
@@ -75,6 +75,7 @@
             crossorigin="anonymous"></script>
     <script src="/campaign/js/app.min.js"></script>
     <script src="/js/das-all.js"></script>
+    <script src="https://localhost:44308/js/cookieConsent.js"></script>
```
[the remark this answers is not in the document]
Comment (dashton82) [reply src/SFA.DAS.Campaign.Web/Views/Shared/_Layout-campaign.cshtml]: This PR has been merged? can it be changed now?

File: src/SFA.DAS.Campaign.Web/wwwroot/js/cookieConsent.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    }
+}
+
+function checkCookie(name) {
+    var nameEQ = name + "=";
+    var ca = document.cookie.split(';');
+    for (var i = 0; i < ca.length; i++) {
+        var c = ca[i];
+        while (c.charAt(0) == ' ') c = c.substring(1, c.length);
+        if (c.indexOf(nameEQ) == 0) return c.substring(nameEQ.length, c.length);
```
Comment (dashton82) [inline src/SFA.DAS.Campaign.Web/wwwroot/js/cookieConsent.js]:
```suggestion
        if (c.indexOf(nameEQ) == 0){ 
        return c.substring(nameEQ.length, c.length);
        }
```

File: src/SFA.DAS.Campaign.Web/wwwroot/js/cookieConsent.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        document.cookie = name + "=" + value + expires + "; path=/";
+    }
+}
+
+function checkCookie(name) {
+    var nameEQ = name + "=";
+    var ca = document.cookie.split(';');
+    for (var i = 0; i < ca.length; i++) {
+        var c = ca[i];
+        while (c.charAt(0) == ' ') c = c.substring(1, c.length);
```
Comment (dashton82) [inline src/SFA.DAS.Campaign.Web/wwwroot/js/cookieConsent.js]:
```suggestion
        while (c.charAt(0) == ' ') {
        c = c.substring(1, c.length);
        }
```

Comment (dashton82) [reply src/SFA.DAS.Campaign.Web/wwwroot/js/cookieConsent.js]: Makes it a bit easier to read 😄

File: src/SFA.DAS.Campaign.Web/wwwroot/js/cookieConsent.js (hunk 2 above)
Comment (dashton82) [reply src/SFA.DAS.Campaign.Web/wwwroot/js/cookieConsent.js]: Makes it a bit easier to read 😄

File: src/SFA.DAS.Campaign.Web/wwwroot/js/cookieConsent.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        var expires = "; expires=" + date.toGMTString();
+    }
+    else var expires = "";
+    if (window.dropCookie) {
+        document.cookie = name + "=" + value + expires + "; path=/";
+    }
+}
+
+function checkCookie(name) {
+    var nameEQ = name + "=";
```
Comment (dashton82) [inline src/SFA.DAS.Campaign.Web/wwwroot/js/cookieConsent.js]: do we need to do this? or is it that cookie names share the same beginning part up to the equals sign?

File: src/SFA.DAS.Campaign.Web/Views/Shared/Components/Form/cookieSettings.cshtml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                <div class="grid-row toggleContainer">
+                    <div class="grid-column-one-quarter">
+
+                        <input type="checkbox" name="cbxAnalyticsConsent" id="cbxAnalyticsConsent" onchange="setChecked(this,'AnalyticsConsent')" checked>
+                        <label for="cbxAnalyticsConsent" data-on="ON" data-off="OFF"></label>
+
+                    </div>
+                    <div class="grid-column-three-quarters">
+                        <h3 class="heading-xs">Analytics cookies</h3>
+                        <p>Help us to improve our website by collecting and reporting information on how it is being used. .</p>
```
Comment (dashton82) [inline src/SFA.DAS.Campaign.Web/Views/Shared/Components/Form/cookieSettings.cshtml]: Extra full stop at the end here

File: src/SFA.DAS.Campaign.Web/Views/Shared/Components/Form/cookieSettings.cshtml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+                You can at any time change or withdraw your consent below
+            </p>
+            <hr />
+            <div class="form-group">
+                <div class="grid-row toggleContainer">
+                    <div class="grid-column-one-quarter">
+                    </div>
+                    <div class="grid-column-three-quarters">
+                        <h3 class="heading-xs">Necessary cookies</h3>
+                        <p> Ensure the basic performance and user friendliness of the website. The website cannot function properly without these cookies, which can only be disabled by changing your browser preferences </p>
```
Comment (dashton82) [inline src/SFA.DAS.Campaign.Web/Views/Shared/Components/Form/cookieSettings.cshtml]: Does this need a full stop at the end?

File: src/SFA.DAS.Campaign.Web/Views/Shared/Components/Form/cookieSettings.cshtml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+            <h1 class="heading-m">Manage cookies on this website</h1>
+            <p>
+                Cookies are small text files that can be used by websites to make a user's experience more efficient.
+            </p>
+            <p>
+                The law states that we can store cookies on your device if they are strictly necessary for the operation of
+                this site. For all other types of cookies we need your permission.
+            </p>
+            <p>
+                You can at any time change or withdraw your consent below
```
Comment (dashton82) [inline src/SFA.DAS.Campaign.Web/Views/Shared/Components/Form/cookieSettings.cshtml]: Full stop?

File: src/SFA.DAS.Campaign.Web/Views/Shared/_Layout-campaign.cshtml (hunk 1 above)
[the remark this answers is not in the document]
Comment (desderry) [reply src/SFA.DAS.Campaign.Web/Views/Shared/_Layout-campaign.cshtml]: i've removed it and the js file as no longer needed with PR being completed

File: src/SFA.DAS.Campaign.Web/Views/Shared/Components/Form/cookieSettings.cshtml (hunk 5 above)
Comment (desderry) [reply src/SFA.DAS.Campaign.Web/Views/Shared/Components/Form/cookieSettings.cshtml]: updated

File: src/SFA.DAS.Campaign.Web/Views/Shared/Components/Form/cookieSettings.cshtml (hunk 6 above)
Comment (desderry) [reply src/SFA.DAS.Campaign.Web/Views/Shared/Components/Form/cookieSettings.cshtml]: yep, ive added it

File: src/SFA.DAS.Campaign.Web/Views/Shared/Components/Form/cookieSettings.cshtml (hunk 7 above)
Comment (desderry) [reply src/SFA.DAS.Campaign.Web/Views/Shared/Components/Form/cookieSettings.cshtml]: added :)

File: src/SFA.DAS.Campaign.Web/wwwroot/js/cookieConsent.js (hunk 2 above)
Comment (desderry) [reply src/SFA.DAS.Campaign.Web/wwwroot/js/cookieConsent.js]: file has gone!

File: src/SFA.DAS.Campaign.Web/wwwroot/js/cookieConsent.js (hunk 3 above)
Comment (desderry) [reply src/SFA.DAS.Campaign.Web/wwwroot/js/cookieConsent.js]: file has gone!

File: src/SFA.DAS.Campaign.Web/wwwroot/js/cookieConsent.js (hunk 4 above)
Comment (desderry) [reply src/SFA.DAS.Campaign.Web/wwwroot/js/cookieConsent.js]: file has gone!

**Pull request merged**